### PR TITLE
feat: support readTime in Datastore query splitter.

### PR DESCRIPTION
**Pull request by yixiaoshen**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-datastore/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

File: datastore-v1-proto-client/src/main/java/com/google/datastore/v1/client/QuerySplitterImpl.java [hunk 1]
```diff
@@ -63,7 +66,23 @@ private QuerySplitterImpl() {
   public List<Query> getSplits(
       Query query, PartitionId partition, int numSplits, Datastore datastore)
       throws DatastoreException, IllegalArgumentException {
+    return getSplitsInternal(query, partition, numSplits, datastore, null);
+  }
 
+  @Override
+  public List<Query> getSplits(
```
Comment (kolea2) [inline datastore-v1-proto-client/src/main/java/com/google/datastore/v1/client/QuerySplitterImpl.java:73]: please add `@BetaApi` in appropriate places

File: datastore-v1-proto-client/src/main/java/com/google/datastore/v1/client/testing/MockDatastoreFactory.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+import com.google.protobuf.Message;
+import com.google.rpc.Code;
+import com.google.rpc.Status;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.List;
+
+/** Fake Datastore factory used for testing purposes when a true Datastore service is not needed. */
+public class MockDatastoreFactory extends DatastoreFactory {
+  int nextStatus;
```
Comment (kolea2) [inline datastore-v1-proto-client/src/main/java/com/google/datastore/v1/client/testing/MockDatastoreFactory.java:43]: nit - please make all these variables `private`

**Review by kolea2 — CHANGES_REQUESTED** [review]

**Review by kolea2 — CHANGES_REQUESTED** [review]
also, we can fix clirr by adding the method to a `clirr-ignored-differences.xml` like https://github.com/googleapis/java-datastore/blob/main/google-cloud-datastore/clirr-ignored-differences.xml (since we added a default method, this is okay)

File: datastore-v1-proto-client/src/main/java/com/google/datastore/v1/client/QuerySplitterImpl.java (hunk 1 above)
Comment (yixiaoshen) [reply datastore-v1-proto-client/src/main/java/com/google/datastore/v1/client/QuerySplitterImpl.java:75]: Done

File: datastore-v1-proto-client/src/main/java/com/google/datastore/v1/client/testing/MockDatastoreFactory.java (hunk 2 above)
Comment (yixiaoshen) [reply datastore-v1-proto-client/src/main/java/com/google/datastore/v1/client/testing/MockDatastoreFactory.java]: Done

File: datastore-v1-proto-client/src/main/java/com/google/datastore/v1/client/testing/MockDatastoreFactory.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+                return response;
+              }
+            };
+          }
+        };
+    Credential credential = options.getCredential();
+    return transport.createRequestFactory(credential);
+  }
+
+  public String lastPath() {
```
[the remark this answers is not in the document]
Comment (yixiaoshen) [reply datastore-v1-proto-client/src/main/java/com/google/datastore/v1/client/testing/MockDatastoreFactory.java]: Done

**Comment by yixiaoshen** [comment]
> also, we can fix clirr by adding the method to a `clirr-ignored-differences.xml` like https://github.com/googleapis/java-datastore/blob/main/google-cloud-datastore/clirr-ignored-differences.xml (since we added a default method, this is okay)

Done

**Review by kolea2 — APPROVED** [review]

**Pull request merged**